homework2/task2.c

```diff
@@ -0,0 +1,89 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+#define uint unsigned int
+
+#define powType float
+
+// #define DEBUG 1
+
+powType slowpow(powType base, uint power)
+{
+    powType ans = 1;
+    for (uint i = 0; i < power; ++i)
+    {
+        ans *= base;
+    }
+    return ans;
+}
+
+powType fastpow(powType base, uint power)
+{
+    uint powerOfTwo = 1;
+
+    powType powerOfBase = base;
+    powType ans = 1;
+
+    while (powerOfTwo <= power)
+    {
+        if (powerOfTwo & power)
+        {
+            ans *= powerOfBase;
+        }
+        powerOfBase *= powerOfBase;
+        powerOfTwo <<= 1;
+    }
+
+    return ans;
+}
+
+#ifdef DEBUG
+int test(void)
```

Comment:
```suggestion
bool test()
```
Здесь значения не имеет - так зачем писать?
Тесты возвращают bool.